src/controllers/utils/queries.js

```diff
@@ -4,6 +4,12 @@ const fullTextSearchQuery = (keyword, isUsingMainKey) => (isUsingMainKey && !key
   ? { word: { $regex: /./ } }
   : { $text: { $search: keyword } }
 );
+
+/* Searching for words that match the keyword and also the wordClass*/
+export const searchIgboTextWithWordClass = (keyword, wordClass, isUsingMainKey) => (isUsingMainKey && !keyword
```

Comment:
Functions with three or more arguments usually are avoided, you can address this by passing in one argument that's an object that has the following keys:

```js
export const searchIgboTextWithWordClass = ({keyword, wordClass, isUsingMainKey }) => {}
```